### ISSUE TITLE
Force derivatives of Model.return_mw_weights() change with each in-line call

### ISSUE THREAD

**Issue by dc-broo3** [sophialilleengen/mwlmc#7]
mwhxforceweights,mwhyforceweights,mwhzforceweights = Model.return_mw_weights(x, y, z)[1:]

With each call in-line of this function for fixed position (x,y,z) the basis weights for the forces change. The potential basis weights remain consistent.

Seems that mwhyforceweights is the only one that changes with each call.

**Comment by michael-petersen** [comment]
Okay, this is definitely a bug. Can you make a minimum working example of the issue and then I'll figure out where this is going wrong?

@sophialilleengen, feel free to assign this to me so that it shows up in my task list (and in general for bugs like this going forward!).

**Comment by dc-broo3** [comment]
Yep, sorry was meant to follow up on this as just flagged it up when I came across it! I've made a short gist here: https://gist.github.com/dc-broo3/75048b41102179a45d5873702a07878a 

Turns out its a bug in the yth MW and LMC halo force weights when calling from return_mw_weights and return_lmc_weights, respectively.

**Comment by michael-petersen** [comment]
Thanks for the MWE! I fixed this bug in the [forked version of the repository](https://github.com/michael-petersen/sphexpansion/commit/a90514940e058523c9ea197ab595a6f97b9c839f): there was a non-initialised variable problem (typical when you see non-repeating behaviour like this). Should be straightforward to fix either by pulling from the fork or just editing the repository directly.
